### PR TITLE
Update monitoring to actually include everything needed

### PR DESCRIPTION
**Pull request by erasche**
Includes:

- setting up: telegraf, influx, grafana
- building graphs
- copying dashboards
- adding alerts
- monitoring with gxadmin

I've marked it "new tutorial" because it might as well be. I wrote the original and it had next to no information, it was useless on its own, without significant extra knowledge. Now all of that is captured in a tutorial you can actually follow on your own.

### CODE REVIEW

File: topics/admin/tutorials/monitoring/tutorial.md [hunk 1]
```diff
@@ -38,43 +46,299 @@ Monitoring is an incredibly important part of server monitoring and maintenance.
 >
 {: .agenda}
 
-# Telegraf
+This tutorial explicitly assumes you are starting from a setup like what was created in the *Galaxy installation with Ansible* tutorial.
```
[the remark this answers is not in the document]
Comment (erasche) [reply topics/admin/tutorials/monitoring/tutorial.md]:
```suggestion
This tutorial explicitly assumes you are starting with a setup like that created in the *Galaxy installation with Ansible* tutorial
```

File: topics/admin/tutorials/monitoring/tutorial.md [hunk 2]
```diff
@@ -38,43 +46,299 @@ Monitoring is an incredibly important part of server monitoring and maintenance.
...
+
+Telegraf communicates with InfluxDB using a JSON based HTTP API, making it easy to write custom tooling to send data there, if need be. Grafana provides the visualisation component, the most interesting component for most consumers of the metrics and monitoring.
+
+
+# Infrastructure
+
+Setting up the infrastructure is quite simple thanks to the automation provided by Ansible. We will first setup a playbook for a "monitoring" machine where our data collection and visualisation, and then we will expand the Galaxy playbook to include Telegraf to monitor the machine and collect data on Galaxy. In this tutorial we will do everything on the same machine. In practice you can separate these services to different machines, if you wish. The only requirement is that Telegraf run on the machine from which you wish to collect data.
```
[the remark this answers is not in the document]
Comment (erasche) [reply topics/admin/tutorials/monitoring/tutorial.md]:
```suggestion
Setting up the infrastructure is quite simple thanks to the automation provided by Ansible. We will first setup a playbook for a "monitoring" machine, which will collect and visualize our data using InfluxDB and Grafana. We will then expand the Galaxy playbook to include Telegraf on the machine to monitor both it and Galaxy itself. In this tutorial we will do everything on the same machine. In practice you can separate these services to different machines, if you wish. The only requirement is that Telegraf run on the machine from which you wish to collect data.
```

File: topics/admin/tutorials/monitoring/tutorial.md [hunk 3]
```diff
@@ -38,43 +46,299 @@ Monitoring is an incredibly important part of server monitoring and maintenance.
...
+
+The available Ansible roles for InfluxDB unfortunately do not support configuring databases or users or retention policies. Ansible itself contains [several modules](https://docs.ansible.com/ansible/latest/modules/list_of_database_modules.html#influxdb) you can use to write your own roles, but nothing generic. UseGalaxy.eu wrote [their own role](https://github.com/usegalaxy-eu/infrastructure-playbook/blob/master/roles/hxr.influxdb/tasks/main.yml) for setting up their database, but this is not reusable enough that it has been split out into a separate role yet. If you plan to automate your entire setup, this can perhaps provide inspiration for writing your own role. Otherwise it is sufficient to manually create your users and retention policies as a one-off task.
```
[the remark this answers is not in the document]
Comment (erasche) [reply topics/admin/tutorials/monitoring/tutorial.md]:
```suggestion
The available Ansible roles for InfluxDB unfortunately do not support configuring databases or users or retention policies. Ansible itself contains [several modules](https://docs.ansible.com/ansible/latest/modules/list_of_database_modules.html#influxdb) you can use to write your own roles, but nothing generic. UseGalaxy.eu wrote [their own role](https://github.com/usegalaxy-eu/infrastructure-playbook/blob/master/roles/hxr.influxdb/tasks/main.yml) for setting up their InfluxDB database, but it is not reusable enough for it to be used here yet. If you plan to automate your entire setup, this tutorial can perhaps provide inspiration for writing your own Ansible role. However, in this case it is sufficient to manually create your users and retention policies as a one-off task.```

File: topics/admin/tutorials/monitoring/tutorial.md [hunk 4]
```diff
@@ -38,43 +46,299 @@ Monitoring is an incredibly important part of server monitoring and maintenance.
...
+>    During this tutorial we will install everything on the same host, but often one keeps the monitoring infrastructure (Grafana, InfluxDB) on a separate host.
+>
+> 4. Edit the inventory file (`hosts`) an add a group for monitoring like:
+>
+>    ```ini
+>    [monitoring]
+>    training-0.example.org ansible_connection=local
+>    ```
+>
+>    Ensure that the hostname is the full hostname of your machine.
```
[the remark this answers is not in the document]
Comment (erasche) [reply topics/admin/tutorials/monitoring/tutorial.md]:
```suggestion
>    **Ensure that the hostname is the full hostname of your machine.**
```

File: topics/admin/tutorials/monitoring/tutorial.md [hunk 5]
```diff
@@ -38,43 +46,299 @@ Monitoring is an incredibly important part of server monitoring and maintenance.
...
+subscriber
+tsm1_cache
+tsm1_engine
+tsm1_filestore
+tsm1_wal
+write
+```
 
-> ### {% icon hands_on %} Hands-on: Telegraf Configuration
+This provides commands like `show databases` and others, but we will not use this interface very often. Telegraf will automatically try to create any database needed, and no interaction is required to setup Grafana to talk to the database.
```
[the remark this answers is not in the document]
Comment (erasche) [reply topics/admin/tutorials/monitoring/tutorial.md]:
```suggestion
The `influx` command provides command line access to InfluxDB in a similar fashion to `psql` for Postgresql. It provides commands like `show databases` and others, but we will not use this interface very often. Telegraf will automatically try to create any database needed, and no interaction is required to setup Grafana to talk to the database.
```

thanks, analogy here is a good idea!

File: topics/admin/tutorials/monitoring/tutorial.md [hunk 6]
```diff
@@ -38,43 +46,299 @@ Monitoring is an incredibly important part of server monitoring and maintenance.
...
+
+## Grafana
+
+[Grafana](https://grafana.com/) provides a visual interface to our metrics. It provides a nice query builder that provides a uniform experience across multiple backend databases, and many attractive graphing options. Another benefit is that many of the UseGalaxy.\* servers share their publish their dashboards publicly, and you can easily copy these and use them on your own server.
```
[the remark this answers is not in the document]
Comment (erasche) [reply topics/admin/tutorials/monitoring/tutorial.md]: definitely needed too, thanks!

**Comment by erasche** [comment]
No worries, all good suggestions, thanks! Should all be addressed now :)

**Comment by shiltemann** [comment]
thanks @erasche! and thanks for the review @Slugger70!

**Pull request merged**